src/main/java/meditracker/medication/MedicationManager.java

```diff
@@ -85,6 +85,97 @@ public static Medication getMedication(String name) throws MedicationNotFoundExc
         throw new MedicationNotFoundException();
     }
 
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication quantity to retrieve medications from the list.
+     * Shows all medications that is equal to or less than specified quantity,
+     *
+     * @param quantity Quantity of the medication to retrieve in double
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByQuantity(Double quantity) throws MedicationNotFoundException {
+        int medicationsFound = 0;
+
+        for (Medication medication : medications) {
+            Double medicationListQuantity = medication.getQuantity();
+            if (Double.compare(medicationListQuantity, quantity) < 0
+                    || Double.compare(medicationListQuantity, quantity) == 0) {
+                medicationsFound++;
+                Ui.printSpecificMed(medication);
+            }
+        }
+        if (medicationsFound == 0) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    /**
+     * Gets the Medication object from the medications list.
+     * Uses the Medication name to retrieve medications from the list.
+     * Shows all medications that contains the name.
+     *
+     * @param name Name of the medication to find and retrieve
+     * @throws MedicationNotFoundException No Medication matching specified name found
+     */
+    public static void showMedicationsByName(String name) throws MedicationNotFoundException {
+        int medicationsFound = 0;
+        for (Medication medication : medications) {
+            if (medication.getName().toLowerCase().contains(name)) {
```

Comment:
Might want to consider adding `name = name.toLowerCase()` before the for loop so you are ignoring case when checking

Reply:
Resolved in [5d3c054](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/206/commits/5d3c054aed17dde1fe3e017b97821d899149787e)